semantic/lookup.go

```diff
@@ -0,0 +1,99 @@
+package semantic
+
+import (
+	flatbuffers "github.com/google/flatbuffers/go"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/libflux/go/libflux"
+	"github.com/influxdata/flux/semantic/internal/fbsemantic"
+)
+
+// LookupBuiltInType returns the type of the builtin value for a given
+// Flux stdlib package. Returns a empty MonoType if lookup fails.
+func LookupBuiltInType(pkg, name string) (MonoType, error) {
+	byteArr := libflux.EnvStdlib()
+	env := fbsemantic.GetRootAsTypeEnvironment(byteArr, 0)
+	var table flatbuffers.Table
+	// perform a lookup of package identifier in flatbuffers TypeEnvironment
+	prop, err := Lookup(env, pkg, name)
+	if err != nil {
+		return MonoType{}, err
+	}
+	if !prop.V(&table) {
+		return MonoType{}, errors.Newf(codes.Internal, "Prop value is not valid: %v", err)
+	}
+	monotype, err := NewMonoType(table, prop.VType())
+	if err != nil {
+		return MonoType{}, err
+	}
+	// return fb polytype within semantic wrapper
+	return monotype, nil
+}
+
+// lookup is a helper function that performs a lookup of a package identifier in
+// a flatbuffers TypeEnvironment. It first checks for the package using path string
+// and then checks for the package identifier, the "name" string in this case, returning
+// the corresponding MonoType if found
+func Lookup(env *fbsemantic.TypeEnvironment, pkg, name string) (*fbsemantic.Prop, error) {
```

Comment:
I needed to make this public since testing was in a different package. I guess I could write a wrapper that makes this accessible from the testing package?